src/java.base/share/classes/java/lang/foreign/GroupLayout.java

```diff
@@ -82,73 +82,81 @@ public sealed interface GroupLayout extends MemoryLayout permits StructLayout, U
-     *        and with an array component type <em>C</em>) that can be mapped to <em>E<sub>b</sub></em> via a resulting
-     *        "array mapper" <em>A2</em> obtained via recursively pealing off nested sequence layouts in <em>E<sub>b</sub></em>
-     *        and then (after <em>D</em> pealing operations)
-     *        finally determining the leaf element layout <em>LL</em> = {@link SequenceLayout#elementLayout() elementLayout()}
-     *        and subsequently obtaining a leaf record mapper:
+     *        <h4>If {@code e_b} is a {@link SequenceLayout };</h4>
+     *        then {@code c_a} must be an array {@code C[]^D} (an array of depth {@code D}
+     *        and with an array component type {@code C}) that can be mapped to {@code e_b} via a resulting
+     *        "array mapper" {@code A2} obtained via recursively pealing off nested sequence layouts in {@code e_b}
+     *        and then (after {@code D} pealing operations)
+     *        finally determining the leaf element layout {@code LL = } {@link SequenceLayout#elementLayout() elementLayout()}
+     *        and subsequently obtaining an array mapper {@code AM(MemorySegment ms, long componentOffset)} with
+     *        a leaf mapper {@code LM}:
      *        <ul>
      *            <li>
-     *            if <em>LL</em> is a {@link ValueLayout}:
-     *            <em>LM</em> = {@link MemorySegment#get(ValueLayout.OfInt, long) ms -> ms.get(LL, offset)}
+     *            if {@code LL} is a {@link ValueLayout}:
+     *            {@code LM} = {@link MemorySegment#get(ValueLayout.OfInt, long) ms -> ms.get(LL, offset + offsetBetween(layout, e_b))}
      *            </li>
      *
      *            <li>
-     *            if <em>LL</em> is a {@link GroupLayout}:
-     *            <em>LM</em> = {@link #recordMapper(Class) LL.recordMapper(C.type())}
+     *            if {@code LL} is a {@link GroupLayout}:
+     *            {@code LM} = M(C.type, LL, offset + offsetBetween(layout, e_b)} recursively.
      *            </li>
      *        </ul>
-     *        whereby <em>C<sub>a</sub></em> = f<sub>a</sub>(MemorySegment ms) will be extracted by
-     *        applying {@code A2} which, in turn, will apply {@code LM} recursively at the appropriate offset(s).<br>
+     *        whereby {@code c_a} = {@code f_a(MemorySegment ms)} will be extracted by
+     *        applying {@code AM} which, in turn, will apply {@code LM} recursively at
+     *        a memory segment slice at the applicable multidimensional array offset(s).<br>
      *        Note: boolean arrays are not supported despite the above and if an attempt is made to map
      *        a boolean array, an {@link IllegalArgumentException} will be thrown.
      *    </li>
      *    <li>
-     *        <h4>If <em>E<sub>b</sub></em> is a {@link PaddingLayout };</h4>
+     *        <h4>If {@code e_b} is a {@link PaddingLayout };</h4>
```

Comment:
It feels like these two last bullets can be merged into "otherwise throw". Maybe that text can make an example of when that could happen (e.g. padding layout or String carrier).